src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -231,6 +233,22 @@ internal void Bind()
                 /*
                  * Build setups
                  */
+                
+                // check to see if we need a LoggerSetup
+                var hasLoggerSetup = config.Setups.Any(c => c is LoggerFactorySetup);
+                if (!hasLoggerSetup)
+                {
+                    var logger = sp.GetService<ILoggerFactory>();
```

Comment:
If the `ILoggerFactory` is available in the `IServiceProvider`, add the setup anyway. Costs nothing.

Reply:
This eliminates the need for any reflection magic and this scenario is covered by the `End2EndSpecs` I added.

Reply:
Oh, makes sense, good idea